npm/private/npm_translate_lock.bzl

```diff
@@ -485,14 +494,42 @@ def npm_translate_lock(
 
             Default: `{dirname}`
 
+        use_starlark_yaml_parser: Opt-out of using `yq` to parse the pnpm-lock file which was added
+            in https://github.com/aspect-build/rules_js/pull/1458 and use the the legacy starlark yaml
+            parser instead.
+
+            This opt-out is a return safety in cases where yq is not able to parse the pnpm generated
+            yaml file. For example, this has been observed to happen due to a line such as the following
+            in the pnpm generated lock file:
+
+            ```
+            resolution: {tarball: https://gitpkg.vercel.app/blockprotocol/blockprotocol/packages/%40blockprotocol/type-system-web?6526c0e}
+            ```
+
+            The `?` character in the `tarball` value causes `yq` to fail with:
+
+            ```
+            $ yq pnpm-lock.yaml -o=json
+            Error: bad file 'pnpm-lock.yaml': yaml: line 7129: did not find expected ',' or '}'
+            ```
+
+            If the tarball value is quoted then yq would accept it but as of this writing, the latest
+            version of pnpm (8.14.3) does not quote such a value and the latest version of yq (4.40.5)
```

Comment:
Is there a `yq` or `pnpm` bug we can link to?

Reply:
Ahh. Thanks for the find @DavidZbarsky-at

Reply:
Added a link to this issue to the docstring here.